test/e2e/kube-ovn/vpc-internal-lb/vpc-internal-lb.go

```diff
@@ -0,0 +1,298 @@
+		}
+		suffix = framework.RandomSuffix()
+
+		namespaceName = f.Namespace.Name
+		slrName = generateSwitchLBRuleName(suffix)
+		svcName = generateServiceName(slrName)
+		podName = generatePodName(suffix)
+
+		var (
+			pod     *corev1.Pod
+			command []string
+			labels  map[string]string
+		)
+		labels = map[string]string{"app": "test"}
+
+		ginkgo.By("Creating pod " + podName)
+		command = []string{"sh", "-c", "sleep infinity"}
+
+		pod = framework.MakePod(namespaceName, podName, labels, nil, image, command, nil)
+		podClient.CreateSync(pod)
+	})
+
+	ginkgo.AfterEach(func() {
+		ginkgo.By("Deleting pod " + podName)
+		podClient.DeleteSync(podName)
+
+		ginkgo.By("Deleting switch-lb-rule " + slrName)
+		switchLBRuleClient.DeleteSync(slrName)
+	})
+
+	framework.ConformanceIt("should create switch-lb-rule with selector for vpc-internal-lb", func() {
+		var (
+			pod *corev1.Pod
+			err error
+		)
+
+		ginkgo.By("Waiting for pod " + podName + " to be ready")
+		framework.WaitUntil(func() (bool, error) {
+			pod, err = podClient.PodInterface.Get(context.TODO(), podName, metav1.GetOptions{})
+			if err == nil {
+				return true, nil
+			}
+			if k8serrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}, fmt.Sprintf("pod %s is created", podName))
+
+		ginkgo.By("Creating SwitchLBRule " + slrName)
+		var (
+			rule                *apiv1.SwitchLBRule
+			selector, endpoints []string
+			ports               []apiv1.SlrPort
+			sessionAffinity     corev1.ServiceAffinity
+			vip                 string
+		)
+
+		vip = "1.1.1.1"
+		sessionAffinity = corev1.ServiceAffinityClientIP
+		ports = []apiv1.SlrPort{
+			{
+				Name:       "dns",
+				Port:       8888,
+				TargetPort: 80,
+				Protocol:   "TCP",
+			},
+		}
+		selector = []string{
+			"app:test",
+		}
+
+		rule = framework.MakeSwitchLBRule(slrName, namespaceName, vip, sessionAffinity, nil, selector, endpoints, ports)
+		_ = switchLBRuleClient.Create(rule)
+
+		ginkgo.By("Waiting for switch-lb-rule " + slrName + " to be ready")
+		framework.WaitUntil(func() (bool, error) {
+			_, err = switchLBRuleClient.SwitchLBRuleInterface.Get(context.TODO(), slrName, metav1.GetOptions{})
+			if err == nil {
+				return true, nil
+			}
+			if k8serrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}, fmt.Sprintf("switch-lb-rule %s is created", slrName))
+
+		var (
+			svc *corev1.Service
+			eps *corev1.Endpoints
+		)
+
+		ginkgo.By("Waiting for headless service " + svcName + " to be ready")
+		framework.WaitUntil(func() (bool, error) {
+			svc, err = serviceClient.ServiceInterface.Get(context.TODO(), svcName, metav1.GetOptions{})
+			if err == nil {
+				return true, nil
+			}
+			if k8serrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}, fmt.Sprintf("service %s is created", svcName))
+		framework.ExpectNotNil(svc)
+
+		ginkgo.By("Checking endpoints")
+		eps = endpointsClient.Get(svcName)
+		framework.ExpectNotNil(eps)
```

Comment:
Instead of checking the endpoints immediately, we should wait for it to be created and to reach the desired state.